### ISSUE TITLE
ardb.swal.log is getting huge

### ISSUE THREAD

**Issue by daddyz** [yinqiwen/ardb#414]
Hi,
The ardb.swal.log is not getting truncated. I am running latest master with 3 slaves connected. 
slave0:ip=XXXXXX,port=16379,state=online,offset=1466688969197,ack_offset=1466688969197,lag=1,o_buffer_size=0,o_buffer_capacity=8192
slave1:ip=XXXXXX,port=16379,state=online,offset=1466688969197,ack_offset=1466688969183,lag=1,o_buffer_size=0,o_buffer_capacity=8192
slave2:ip=XXXXXX,port=16379,state=online,offset=1466688969197,ack_offset=1466688969183,lag=1,o_buffer_size=0,o_buffer_capacity=8192

The size of the file is 136G, while the data dir is only 15G.
How can I truncate the file and to limit its size?

Here are the versions:
# Server
ardb_version:0.9.6
redis_version:2.6.0
engine:rocksdb
ardb_home:/var/lib/ardb
os:Linux 4.14.12-x86_64-linode92 x86_64

# Databases
data_dir:/var/lib/ardb/data
used_disk_space:15067928396
rocksdb_version:5.9.2
rocksdb.block_table_usage:536870912
rocksdb.block_table_pinned_usage:0
rocksdb_memtable_total:1115239872
rocksdb_memtable_unflushed:1115239872
rocksdb_table_readers_total:793426982
rocksdb.estimate-table-readers-mem:0
rocksdb.cur-size-all-mem-tables:58720992

**Comment by yinqiwen** [comment]
the `ardb.swal.log` size is defiled in conf `repl-backlog-size`.

**Comment by daddyz** [comment]
this was the setting
repl-backlog-size           1000m

I changed it to 10m now, it's still the same

**Comment by yinqiwen** [comment]
the created file `ardb.swal.log` is fixed length file , if u change  `repl-backlog-size` to a smaller value, u need delete the `ardb.swal.log` & `ardb.swal.log.meta` and restart the server.

**Comment by daddyz** [comment]
1000m is 1G of data or 1000m changes?

**Comment by daddyz** [comment]
yes, file was increasing, now I removed the file, restarted ardb and it is still has 10m size

**Comment by yinqiwen** [comment]
is the `ardb.swal.log` opend by 2 or more ardb instances?       

this file is a ring buffer file which would have a fix length configured by `repl-backlog-size`.  the single write thread would change the file position to 0 if the file position reach the length limit .  
 https://github.com/yinqiwen/ardb/blob/0.9/src/repl/swal.c#L201size

**Comment by daddyz** [comment]
Anyway, currently the file stays the same, I will keep monitoring and will let you know in case it will grow.
I didn't catch the time when it started growing, might be because of version change, we did upgrade several weeks ago.
